src/main/java/fi/metatavu/kuntaapi/server/rest/ptv7adapter/Ptv7Adapter.java

```diff
@@ -0,0 +1,260 @@
+  private <T> T translateEntity(Class<T> entityClass, T entity, Map<String, Map<String, String>> entityMapping) throws IOException {
+    if (entity == null) {
+      return null;
+    }
+    
+    T result = cloneEntity(entityClass, entity);
+    
+    entityMapping.keySet().forEach((propertyPath) -> {
+      Map<String, String> propertyMapping = entityMapping.get(propertyPath);
+      try {
+        applyPropertyValue(result, new ArrayList<>(Arrays.asList(StringUtils.split(propertyPath, "/"))), propertyMapping);
+      } catch (IllegalAccessException | InvocationTargetException | IntrospectionException e) {
+        logger.log(Level.WARNING, "Failed to apply PTV 7 compatibility changes", e);
+      }
+    });
+    
+    return result;
+  }
+  
+  /**
+   * Clones entity
+   * 
+   * @param entityClass entity class
+   * @param entity entity
+   * @return cloned entity
+   * @throws IOException thrown when entity reading fails
+   */
+  private <T> T cloneEntity(Class<T> entityClass, T entity) throws IOException {
+    if (entity == null) {
+      return null;
+    }
+    
+    ObjectMapper objectMapper = getObjectMapper();
+    return objectMapper.readValue(objectMapper.writeValueAsBytes(entity), entityClass);
+  }
+
+  /**
+   * Applies PTV8 -> PTV7 naming convention into an object 
+   * 
+   * @param bean object
+   * @param propertyPath property path
+   * @param propertyMapping property mapping 
+   * @throws IllegalAccessException throw on bean read error
+   * @throws InvocationTargetException throw on bean read error
+   * @throws IntrospectionException throw on bean read error
+   */
+  private void applyPropertyValue(Object bean, List<String> propertyPath, Map<String,  String> propertyMapping) throws IllegalAccessException, InvocationTargetException, IntrospectionException {
+    try {
+      String propertyName = propertyPath.remove(0);
+
+      PropertyDescriptor propertyDescriptor = new PropertyDescriptor(propertyName, bean.getClass());
+      Method readMethod = propertyDescriptor.getReadMethod();
+      Method writeMethod = propertyDescriptor.getWriteMethod();
+      
+      if (propertyPath.isEmpty()) {
+        String currentValue = (String) readMethod.invoke(bean);
+        if (currentValue == null) {
+          return;
+        }
+        
+        String newValue = propertyMapping.get(currentValue);
+        if (newValue != null) {
+          writeMethod.invoke(bean, newValue); 
+        }
+      } else {
+        Object value = readMethod.invoke(bean);
+       
+        if (value instanceof List<?>) {
+          List<?> list = (List<?>) value;
+          for (Object item : list) {
+            applyPropertyValue(item, new ArrayList<>(propertyPath), propertyMapping);
+          }
+        } else {
+          applyPropertyValue(value, new ArrayList<>(propertyPath), propertyMapping);
+        }
+      }
+    } catch (IllegalAccessException | InvocationTargetException | IntrospectionException e) {
+      e.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)